javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,453 @@
+
+    changed
+  }
+
+  def removeFirstOccurrence(o: Any): Boolean =
+    _removeOccurrence(iterator(), o)
+
+  def removeLastOccurrence(o: Any): Boolean =
+    _removeOccurrence(descendingIterator(), o)
+
+  override def listIterator(index: Int): ListIterator[E] = {
+    checkIndexOnBounds(index)
+    //new RandomAccessListIterator(this, index, 0, _size)
+    new ListIterator[E] {
+
+      private var last = -1
+      private var i = index
+
+      private var currentNode: Node[Box[E]] =
+        if (index == size) null
+        else getNodeAt(index)
+
+      private var lastNode: Node[Box[E]] =
+        if (currentNode eq null) getNodeAt(size - 1)
+        else null
+
+      def hasNext(): Boolean =
+        (i < size)
+
+      def next(): E = {
+        if (i > size) throw new NoSuchElementException()
+        last = i
+        i += 1
+
+        if (currentNode eq null)
+          currentNode = lastNode
+
+        lastNode = currentNode
+        currentNode = currentNode.next
+
+        lastNode.value.inner
+      }
+
+      def hasPrevious(): Boolean =
+        (i > 0)
+
+      def previous(): E = {
+        if (i < 0) throw new NoSuchElementException()
+        i -= 1
+        last = i
+
+        if (currentNode eq null)
+          currentNode = lastNode
+        else
+          currentNode = currentNode.prev
+
+        lastNode = currentNode
+
+        lastNode.value.inner
+      }
+
+      def nextIndex(): Int = i
+
+      def previousIndex(): Int = i - 1
+
+      def remove(): Unit = {
+        checkThatHasLast()
+        removeNode(lastNode)
+        if (last < i)
+          i -= 1
+        last = -1
+      }
+
+      def set(e: E): Unit = {
+        checkThatHasLast()
+        lastNode.value = Box(e)
+      }
+
+      def add(e: E): Unit = {
+        if (currentNode eq null) {
+          if (lastNode.next eq null) {
+            addLast(e)
+            lastNode = getNodeAt(size - 1)
+          } else if (lastNode.prev eq null) {
+            addFirst(e)
+            lastNode = getNodeAt(0)
+          }
+        } else {
+          addNode(currentNode, e)
+        }
+        last = -1
+        i += 1
+      }
+
+      private def checkThatHasLast(): Unit = {
+        if (last == -1)
+          throw new IllegalStateException()
+      }
+    }
+  }
+
+  def descendingIterator(): Iterator[E] = {
+    new Iterator[E] {
+      private val iter = listIterator(_size)
+
+      def hasNext(): Boolean =
+        iter.hasPrevious
+
+      def next(): E =
+        iter.previous
+
+      def remove(): Unit =
+        iter.remove
+
+    }
+  }
+
+  override def clone(): AnyRef =
+    new LinkedList[E](this)
+
+  override def toString(): String = {
+    var current = head
+    val str = new StringBuilder()
+
+    str ++= "["
+    while ((current ne null) && (current.value ne null)) {
+      str ++= current.value.inner.toString
+      if (current.next ne null)
+        str ++= ", "
+      current = current.next
+    }
+    str ++= "]"
+    str.result
```

Comment:
No need to override `toString()`. The implementation in `AbstractCollection` is already good.